rlog/2024-12-23-de-mls.mdx

```diff
@@ -0,0 +1,252 @@
+- `welcome`: A special topic for joining the group. With it, we solve the problem of how users can join a group without the server side. This topic is also used for providing the key packages by the users. 
+
+This architecture does not require a direct 1-to-1 channel with the admin for join requests. However, it does allow for the possibility that the admin can personally request a key package and manually add a member to the group.
+
+### Managing the Key Packages
+
+As previously mentioned, the core concept is that users provide their key packages themselves through the welcome topic when joining a group. The working process of the welcome topic is as follows: 
+
+- When a group is created, the admin (or admins) periodically publish a special public key along with a signature at a set frequency (e.g., once per epoch). New users initially subscribe only to this content topic, and upon receiving the message, they encrypt their key package using the provided public key.
+- At the end of the epoch, the administrator prepares a commit message containing the details of all new users (if they are approved to join the group). This commit message is published in the commit channel, followed by a welcome message sent to all new users in the current channel.
+- If the previous process is completed successfully, all new users are subscribed to the relevant topics and receive the group key, which enables encrypted messaging for all members of the group. 
+- Admin deletes the key packages with the next epoch.
+- If a user fails to send a valid message to the current epoch, his message is rejected and he has to get the key again and repeat the whole procedure.
+
+The current implementation assumes support for a single admin. This is because achieving consensus among multiple users would require agreement on which commits to accept, which to reject, and in what order they should be processed. 
+
+## Architecture notes
+
+The architecture of the de-MLS is based on the actor-based model. In this section we will give a brief overview of the architecture and the main actors. 
+Also, we discuss the cryptographic keys and their usage in the implementation. 
+
+### Actor-based model
+
+**Disclaimer:** these architectural solutions are primarily intended to demonstrate the performance of de-mls. 
+Please do not use them in production, they may contain vulnerabilities and may not scale well!
+
+The architecture is built on an actor-based model, enabling a focus on the processing of messages rather than the specifics of how messages are transmitted between different entities. 
+We will give a brief overview of the architecture and the main actors. Let's start with the actors which are provide all network communication: the `waku actor` and the `websocket actor`. 
+
+#### Communication between actors: WebSocket <-> Application <-> Waku
+
+** Websocket actor **
```

Comment:
Does the transport matter? Isnt it just the "app actor"?

Reply:
Do you mean web socket part?

Reply:
yes, I find the "websocket" part confusing